subsys/mgmt/mcumgr/grp/settings_mgmt/src/settings_mgmt.c

```diff
@@ -0,0 +1,535 @@
+	enum mgmt_cb_return status;
+	int32_t ret_rc;
+	uint16_t ret_group;
+
+	/* Send request to application to check if access should be allowed or not */
+	status = mgmt_callback_notify(MGMT_EVT_OP_SETTINGS_MGMT_ACCESS, &settings_access_data,
+				      sizeof(settings_access_data), &ret_rc, &ret_group);
+
+	if (status != MGMT_CB_OK) {
+		bool ok;
+
+		if (status == MGMT_CB_ERROR_RC) {
+			return ret_rc;
+		}
+
+		ok = smp_add_cmd_ret(zse, ret_group, (uint16_t)ret_rc);
+		goto end;
+	}
+#endif
+
+	settings_commit();
+
+#if defined(CONFIG_MCUMGR_GRP_SETTINGS_ACCESS_HOOK)
+end:
+#endif
+	return ok ? MGMT_ERR_EOK : MGMT_ERR_EMSGSIZE;
+}
+
+/**
+ * Command handler: settings load
+ */
+static int settings_mgmt_load(struct smp_streamer *ctxt)
+{
+	bool ok = true;
+
+#if defined(CONFIG_MCUMGR_GRP_SETTINGS_ACCESS_HOOK)
+	struct settings_mgmt_access settings_access_data = {
+		.access = SETTINGS_ACCESS_LOAD,
+	};
+
+	zcbor_state_t *zse = ctxt->writer->zs;
+	enum mgmt_cb_return status;
+	int32_t ret_rc;
+	uint16_t ret_group;
+
+	/* Send request to application to check if access should be allowed or not */
+	status = mgmt_callback_notify(MGMT_EVT_OP_SETTINGS_MGMT_ACCESS, &settings_access_data,
+				      sizeof(settings_access_data), &ret_rc, &ret_group);
+
+	if (status != MGMT_CB_OK) {
+		bool ok;
+
+		if (status == MGMT_CB_ERROR_RC) {
+			return ret_rc;
+		}
+
+		ok = smp_add_cmd_ret(zse, ret_group, (uint16_t)ret_rc);
+		goto end;
+	}
+#endif
+
+	settings_load();
+
+#if defined(CONFIG_MCUMGR_GRP_SETTINGS_ACCESS_HOOK)
+end:
+#endif
+	return ok ? MGMT_ERR_EOK : MGMT_ERR_EMSGSIZE;
+}
+
+/**
+ * Command handler: settings save
+ */
+static int settings_mgmt_save(struct smp_streamer *ctxt)
+{
+	bool ok = true;
+
+#if defined(CONFIG_MCUMGR_GRP_SETTINGS_ACCESS_HOOK)
+	struct settings_mgmt_access settings_access_data = {
+		.access = SETTINGS_ACCESS_SAVE,
+	};
+
+	zcbor_state_t *zse = ctxt->writer->zs;
+	enum mgmt_cb_return status;
+	int32_t ret_rc;
+	uint16_t ret_group;
+
+	/* Send request to application to check if access should be allowed or not */
+	status = mgmt_callback_notify(MGMT_EVT_OP_SETTINGS_MGMT_ACCESS, &settings_access_data,
+				      sizeof(settings_access_data), &ret_rc, &ret_group);
+
+	if (status != MGMT_CB_OK) {
+		bool ok;
+
+		if (status == MGMT_CB_ERROR_RC) {
+			return ret_rc;
+		}
+
+		ok = smp_add_cmd_ret(zse, ret_group, (uint16_t)ret_rc);
+		goto end;
+	}
+#endif
+
+	settings_save();
+
+#if defined(CONFIG_MCUMGR_GRP_SETTINGS_ACCESS_HOOK)
+end:
+#endif
+	return ok ? MGMT_ERR_EOK : MGMT_ERR_EMSGSIZE;
+}
+
+#ifdef CONFIG_MCUMGR_SMP_SUPPORT_ORIGINAL_PROTOCOL
+static int settings_mgmt_translate_error_code(uint16_t ret)
+{
+	int rc;
+
+	switch (ret) {
+	case SETTINGS_MGMT_RET_RC_KEY_TOO_LONG:
+	rc = MGMT_ERR_EINVAL;
+	break;
+	case SETTINGS_MGMT_RET_RC_KEY_NOT_FOUND:
+	case SETTINGS_MGMT_RET_RC_READ_NOT_SUPPORTED:
+	rc = MGMT_ERR_ENOENT;
+	break;
+	case SETTINGS_MGMT_RET_RC_UNKNOWN:
+	default:
+	rc = MGMT_ERR_EUNKNOWN;
+	}
```

Comment:
Is it github incorrectly doing indentations?

Reply:
It is not, fixed